### PR TITLE
Actor squash

### PR DESCRIPTION
**Pull request by jacquese**
TL;DR: combines https://github.com/spatialos/UnrealGDK/pull/2813 and https://github.com/spatialos/UnrealGDK/pull/2701 with final fixes and rebased on master

### The story so far

https://github.com/spatialos/UnrealGDK/pull/2701 is the original Actor system PR. It includes the bulk of the changes in this PR, including the gutting of spatial receiver and creation of the new Actor system class. This class is backed by a subview rather than reading the ops directly, which gives advantages, such as the fact that most of the pending ops data structures previously in the receiver are no longer required, because you only see entities at the point where we would normally process the pending ops anyway.

https://github.com/spatialos/UnrealGDK/pull/2701 worked in that you could open the handover gym, and the cubes would happily move around, which typically means everything is a-ok when making an Unreal GDK change. _However,_ this was not even close to reality. On running the integration tests included in the test gyms repo, many tests failed. A lot of bug fixing has gone in to get it to this point where it now passes CI, including:
- switch statements should have breaks if you don't want them to execute the next case (who knew)
- the metadata component is **not the same** as the unreal metadata component
- removing a component when it shouldn't be (for example in a refresh) results in the whole _Actor_ being marked for destruction and everything starts to null out and your Actor gets load balanced to the wrong worker which is horrible
- tombstones are no longer processed by Actor logic, as the existence of the tombstone component fails the predicate- this is fine, as long as the entity didn't have a tombstone _on checkout_. For that case we introduce a new subview which looks for a new query tag, the Tombstone Tag, which is different to the tombstone component, and on checking out a complete entity by that subview delete the startup actor it corresponds to.

Then, there was another failure- the Debug Interface Test. I never debugged this one. Because! @ImprobableNic's change, changing the debug context to be backed by a subview, fixed it. So that PR, https://github.com/spatialos/UnrealGDK/pull/2813, is combined into this one. Both should go in anyway, as we are aiming to have nowhere that reads information from spatial not backed by a subview.

This PR combines all of the above into a neat, oversized package which is squashed to make it easy to rebase onto master. The separate PRs are more granular and have more info on the individual change, but this is the one that needs to be merged.

Reviewers: @ImprobableNic @samiwh @m-samiec 

https://improbableio.atlassian.net/browse/UNR-4199

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -406,23 +409,33 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	const FFilterPredicate ActorFilter = [](const Worker_EntityId, const SpatialGDK::EntityViewElement& Element) {
 		return !Element.Components.ContainsByPredicate(SpatialGDK::ComponentIdEquality{ SpatialConstants::TOMBSTONE_COMPONENT_ID });
 	};
+	const FFilterPredicate TombstoneFilter = [](const Worker_EntityId, const SpatialGDK::EntityViewElement& Element) {
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:412]: Where is this used ?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -1842,12 +1855,21 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
...
+		{
+			ActorSystem->Advance();
+		}
+
 		{
 			SCOPE_CYCLE_COUNTER(STAT_SpatialProcessOps);
 			Dispatcher->ProcessOps(GetOpsFromEntityDeltas(Connection->GetEntityDeltas()));
 			Dispatcher->ProcessOps(Connection->GetWorkerMessages());
 			CrossServerRPCHandler->ProcessMessages(Connection->GetWorkerMessages(), DeltaTime);
-			Receiver->ProcessActorsFromAsyncLoading();
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1850]: What happens to async loading? Disabled for now?
Or is it a regression from a fix that was done to separate async loading to collect loaded packages in an intermediate array?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 3]
```diff
@@ -0,0 +1,1833 @@
...
+	if (RetiredActorIndex != INDEX_NONE)
+	{
+		EntitiesToRetireOnAuthorityGain.RemoveAtSwap(RetiredActorIndex);
+	}
+
+	if (NetDriver->IsServer())
+	{
+		// Check to see if we are removing a system entity for a worker connection. If so clean up the ClientConnection to delete any and
+		// all actors for this connection's controller.
+		if (FString* WorkerName = WorkerConnectionEntities.Find(EntityId))
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:612]: Should be moved somewhere else, this would break Client disconnection detection otherwise.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:412]: should have been used in the tombstone subview- will update (although won't break anything, just means too many entities are being passed through that subview)

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1850]: yep, disabled. async loading won't work until https://improbableio.atlassian.net/browse/UNR-4198 is done

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 3 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:612]: yeah, it should be. A new subsystem for system entities is a good idea, and we can probably use the system component as a tag. Will add this as a follow up ticket

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:412]: actually no! The reason it was removed from the subview is because there's two different types of tombstoning- so we added a new tag instead, and made that the query tag for the subview. Deleting the filter instead

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: removed now

**Comment by jacquese** [comment]
@ImprobableNic I've now reverted the changes to the debug component to be as your PR had it as well

**Review by ImprobableNic — APPROVED** [review]
I think I called out everything I could find.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1850]: Was there an async loading gym/test @ImprobableNic?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -1689,6 +1699,11 @@ int32 USpatialNetDriver::ServerReplicateActors(float DeltaSeconds)
 		return 0;
 	}
 
+	if (DebugCtx)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1702]:
```suggestion
	if (DebugCtx != nullptr)
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 5]
```diff
@@ -1842,12 +1852,21 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
 			RPCService->AdvanceView();
 		}
 
+		if (DebugCtx != nullptr)
+		{
+			DebugCtx->AdvanceView();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1857]: nit: advance and advanceView are the same, could we be consistent with function names in subviews?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 6]
```diff
@@ -1860,6 +1879,28 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
 			WellKnownEntitySystem->Advance();
 		}
 
+		if (SpatialDebugger != nullptr)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1882]: Shouldn't the SpatialDebugger also use the actorsubview?
Otherwise the debugger could attempt to refer to actors that aren't present.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 7]
```diff
@@ -413,16 +416,23 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 		Connection->GetCoordinator().CreateSubView(SpatialConstants::ACTOR_AUTH_TAG_COMPONENT_ID, ActorFilter, RefreshCallbacks);
 	const SpatialGDK::FSubView& ActorNonAuthSubview =
 		Connection->GetCoordinator().CreateSubView(SpatialConstants::ACTOR_NON_AUTH_TAG_COMPONENT_ID, ActorFilter, RefreshCallbacks);
+	const SpatialGDK::FSubView& TombstoneActorSubview = Connection->GetCoordinator().CreateSubView(
+		SpatialConstants::TOMBSTONE_TAG_COMPONENT_ID, SpatialGDK::FSubView::NoFilter, SpatialGDK::FSubView::NoDispatcherCallbacks);
+	DebugActorSubView =
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: What's the benefit of the netdriver maintain a reference to the debug actor subview? I'd rather the debug ctx subview owns it, similar to how the other subviews function.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 6 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1882]: It should. I didn't want to add that to this PR. It's fairly bloated as is- but I can do it if you feel it's necessary

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 4 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1702]: done

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 5 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1857]: in this case it actually isn't the same! `AdvanceView` calls (both here and the RPC service one) only update their local views, and run no logic. An `Advance` on the other hand (in the case of the actor system and well known entity system) may update state and run logic.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 7 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: good catch- it shouldn't be a member of the net driver

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: it turns out this is actually horrific

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: debug context is initialised by passing a static function the net driver and this appears to be able to happen multiple times- when it succeeds the static function sets a field of the net driver to be an instance of the debug context which is also instantiated inside the static function (???) because otherwise it is possible to enable debugging too soon?

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: my current plan is to rejig this such that: 
- the debug context is created with all the other core classes (if the world settings allows - if anyone knows why the world settings can't be accessed there let me know)
- the subview is passed in at this point if it debug is enabled
- the context is _initialised_ during startup to make sure the LB strategy is ready if it was created (if not nullptr etc.)

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:421]: I did this, and no, the world settings can't be accessed as the world isn't initialised at this point.

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: alright it now owns the subview, but there is other follow on work that needs to be done here- I didn't realise it was using a non-tag component for the subview query tag, and should probably use the actor non-auth tag with a filter for the gdk debug component, because otherwise authority over the debug component might result in checking out an entity before the whole query result is received. It's such an awkward system that I think that is for a future PR

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 8]
```diff
@@ -403,31 +406,37 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
...
 	Dispatcher->Init(Receiver, StaticComponentView, SpatialMetrics, SpatialWorkerFlags);
 	Sender->Init(this, &TimerManager, RPCService.Get(), Connection->GetEventTracer());
-	Receiver->Init(this, &TimerManager, RPCService.Get(), Connection->GetEventTracer());
+	Receiver->Init(this, Connection->GetEventTracer());
 	GlobalStateManager->Init(this);
 	SnapshotManager->Init(Connection, GlobalStateManager, Receiver);
 	PlayerSpawner->Init(this);
 	PlayerSpawner->OnPlayerSpawnFailed.BindUObject(GameInstance, &USpatialGameInstance::HandleOnPlayerSpawnFailed);
 	SpatialMetrics->Init(Connection, NetServerMaxTickRate, IsServer());
+	SpatialMetrics->Init(Connection, NetServerMaxTickRate, IsServer());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:439]: Duplicate line

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 6 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1881]: That's fine, but can we add a ticket please

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverDebugContext.cpp [hunk 9]
```diff
@@ -142,24 +205,44 @@ void USpatialNetDriverDebugContext::RemoveActorTag(AActor* Actor, FName Tag)
 	}
 }
 
-void USpatialNetDriverDebugContext::OnDebugComponentUpdateReceived(Worker_EntityId Entity)
+void USpatialNetDriverDebugContext::AddComponent(Worker_EntityId EntityId)
 {
-	SpatialGDK::DebugComponent* DbgComp = NetDriver->StaticComponentView->GetComponentData<SpatialGDK::DebugComponent>(Entity);
+	const SpatialGDK::EntityViewElement& Element = NetDriver->Connection->GetView().FindChecked(EntityId);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverDebugContext.cpp:210]: Why are we using the connection view and not the subview here?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 10]
```diff
@@ -0,0 +1,1833 @@
...
+
+	for (const EntityDelta& Delta : SubView->GetViewDelta().EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::UPDATE:
+		{
+			for (const AuthorityChange& Change : Delta.AuthorityLostTemporarily)
+			{
+				AuthorityLost(Delta.EntityId, Change.ComponentSetId);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:120]: Shouldn't this be followed by an AuthGained() ?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:120]: Oh I see it
I think i'm gonna be sick
Could we add a log why this is necessary?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 11]
```diff
@@ -0,0 +1,1833 @@
...
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+void ActorSystem::ApplyComponentAdd(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId, Schema_ComponentData* Data)
+{
+	switch (ComponentId)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:208]: qq - where is the rest of the actor data stored?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 12]
```diff
@@ -0,0 +1,1833 @@
...
+
+void ActorSystem::ComponentAdded(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId, Schema_ComponentData* Data)
+{
+	if (ComponentId == SpatialConstants::DORMANT_COMPONENT_ID)
+	{
+		HandleDormantComponentAdded(EntityId);
+		return;
+	}
+
+	if (ComponentId < SpatialConstants::STARTING_GENERATED_COMPONENT_ID
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:416]: Looks like this replace `MAX_RESERVED_SPATIAL_SYSTEM_COMPONENT_ID`, what was the reasoning? Did we have instances where this was happening, or is this just a cleanup?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 13]
```diff
@@ -0,0 +1,1833 @@
...
+		FSpatialGDKSpanId CauseSpanId = EventTracer->GetSpanId(EntityComponentId(EntityId, ComponentId));
+		EventTracer->TraceEvent(FSpatialTraceEventBuilder::CreateComponentUpdate(Channel->Actor, TargetObject, EntityId, ComponentId),
+								CauseSpanId.GetConstId(), 1);
+	}
+
+	ESchemaComponentType Category = NetDriver->ClassInfoManager->GetCategoryByComponentId(ComponentId);
+
+	if (Category == SCHEMA_Data || Category == SCHEMA_OwnerOnly)
+	{
+		// SCOPE_CYCLE_COUNTER(STAT_ReceiverApplyData);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:526]: Ticket for adding these back in (or just add them back in, should be super simple)

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 8 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:439]: great catch, f is too close to d

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverDebugContext.cpp (hunk 9 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverDebugContext.cpp:210]: For the actual answer, when addressing above comments I was surprised that we would grab a view through the netdriver that wasn't through the connection. So rather than clocking that there was a subview owned by the netdriver and fixing that I changed it to connection. Then I went back and fixed the subview thing and left this. Change to the subview

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 10 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:120]: ha
sure yeah good idea, added a comment

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 11 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:208]: what's the rest of the actor data? generated components and things? the actor system only maintains the components it needs the deserialized from of to do its work, anything else is handled as before

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 12 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:416]: hmm. I remember rewriting this bit generally, and it was hard to parse. I went with this because I found it easy to read that if it gets past this point, that means it's a) generated and b) not QBI

can change it back if you feel strongly

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 13 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:526]: "super simple" i'll see if I can easily add these back

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp [hunk 14]
```diff
@@ -0,0 +1,1833 @@
...
+	Schema_Object* FieldsObject = Schema_GetComponentUpdateFields(Update);
+	if (Schema_GetBoolCount(FieldsObject, SpatialConstants::HEARTBEAT_CLIENT_HAS_QUIT_ID) > 0
+		&& GetBoolFromSchema(FieldsObject, SpatialConstants::HEARTBEAT_CLIENT_HAS_QUIT_ID))
+	{
+		// Client has disconnected, let's clean up their connection.
+		CloseClientConnection(NetConnection, EntityId);
+	}
+}
+
+USpatialActorChannel* ActorSystem::GetOrRecreateChannelForDormantActor(AActor* Actor, const Worker_EntityId EntityID) const
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:1189]: I prefer my incorrect spelling

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 13 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp]: added back the 5 that were here and commented out, renamed for this system

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 12 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:425]: Nah it's fine - was more interested in the change

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp (hunk 14 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp:1195]: if it wouldn't have my name next to in the blame, I'd leave it

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp [hunk 15]
```diff
@@ -44,30 +44,24 @@ void SpatialDispatcher::ProcessOps(const TArray<Worker_Op>& Ops)
...
-			Receiver->DropQueuedRemoveComponentOpsForEntity(Op.op.remove_entity.entity_id);
 			break;
 
 		// Components
 		case WORKER_OP_TYPE_ADD_COMPONENT:
 			StaticComponentView->OnAddComponent(Op.op.add_component);
-			Receiver->OnAddComponent(Op.op.add_component);
 			break;
 		case WORKER_OP_TYPE_REMOVE_COMPONENT:
-			Receiver->OnRemoveComponent(Op.op.remove_component);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp:66]: Did we previously never remove individual components from the scv??

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp [hunk 16]
```diff
@@ -78,9 +72,11 @@ void SpatialDispatcher::ProcessOps(const TArray<Worker_Op>& Ops)
 			Receiver->OnCommandResponse(Op);
 			break;
 
-		// Authority Change
+			// Authority Change
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp:75]:
```suggestion
		// Authority Change
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp (hunk 15 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp:66]: Oh, it's done in the receiver

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 17]
```diff
@@ -300,7 +283,11 @@ void ASpatialDebugger::LoadIcons()
 
 void ASpatialDebugger::OnEntityAdded(const Worker_EntityId EntityId)
 {
-	check(NetDriver != nullptr && !NetDriver->IsServer());
+	check(NetDriver != nullptr);
+	if (NetDriver->IsServer())
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:287]: These seem easier and cheaper to check at calling sight, but I guess this is gonna be reworked anyway

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp (hunk 15 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp:66]: aha yeah it used to be in the receiver

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverDebugContext.h [hunk 18]
```diff
@@ -22,16 +23,21 @@ class USpatialNetDriver;
...
+struct ComponentChange;
+}
+
 UCLASS()
 class SPATIALGDK_API USpatialNetDriverDebugContext : public UObject
 {
 	GENERATED_BODY()
 public:
-	static void EnableDebugSpatialGDK(USpatialNetDriver* NetDriver);
+	static void EnableDebugSpatialGDK(const SpatialGDK::FSubView& InSubView, USpatialNetDriver* NetDriver);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverDebugContext.h:36]: Can forward declare this to prevent include

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp (hunk 17 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:287]: yes this has to be rejigged soonish anyway

**Review by m-samiec — APPROVED** [review]
Great work!

**Review by samiwh — APPROVED** [review]
Paired on a fair amount of this. Haven't completely gone through the rest of it but so far it looks good to me.

I think the remainder is sufficiently isolated that it's safer to merge and iterate.
